test/adapters/kyc-onboarding.test.js

```diff
@@ -418,11 +454,131 @@ describe("Adapter - KYC Onboarding", () => {
 
     // Attempt to rejoin the DAO using the same KYC coupon
     await expect(
-      onboarding.onboardEth(dao.address, applicant, signature, {
+      onboarding.onboardEth(dao.address, applicant, memberNonce, signature, {
         from: applicant,
         value: ethAmount,
         gasPrice: toBN("0"),
       })
     ).to.be.revertedWith("already redeemed");
   });
+
+  it("should be possible to first join when configured with canTopUp", async () => {
```

Comment:
We probably need a test case where the member joins for the first time and the top up is disabled. It is related to the use case I mentioned to you on the `KycOnboarding.sol` review.